g11n-ws/modules/md-restful-api-i18n/src/main/java/com/vmware/vip/i18n/api/base/BaseAction.java

```diff
@@ -8,13 +8,44 @@
 
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
 
 import com.vmware.vip.common.i18n.dto.response.APIResponseDTO;
+import com.vmware.vip.common.i18n.status.APIResponseStatus;
 import com.vmware.vip.common.i18n.status.Response;
+import com.vmware.vip.core.messages.service.product.IProductService;
+import com.vmware.vip.i18n.api.base.utils.VersionMatcher;
 
 public class BaseAction {
 	private static Logger logger = LoggerFactory.getLogger(BaseAction.class);
 
+	@Autowired
+	protected IProductService baseProductService;
+	
+	/**
+	 *The method use to get the available version string by matching support versions. 
+     *if not matching return the request version string
+	 */
+	protected String getAvailableVersion(String productName, String version) {
+        try {
+          return VersionMatcher.getMatchedVersion(version, baseProductService.getSupportVersionList(productName));
+        }catch(Exception e) {
+          return version;
+        }
+    }
+   
+	/**
+	 * This method use to handle and package the version fallback response content 
+	 * 
+	 */
+	protected APIResponseDTO handleVersionFallbackResponse(String oldVersion, String newVersion, Object data) {
```

Comment:
How about changing name to handleTranslationResponse?

Reply:
the handleVersionFallbackResponse method not only use translation response but also use the get support language API and get component API, So the name changes to handleTranslationResponse is inappropriate.